src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/Maintainability/UseNameofInPlaceOfString.cs

```diff
@@ -60,16 +60,11 @@ private void AnalyzeArgument(OperationAnalysisContext context)
                 return;
             }
 
-            if (argument.Parameter == null)
-            {
-                return;
-            }
-
             var stringText = (string)argument.Value.ConstantValue.Value;
 
-            var matchingParameter = argument.Parameter;
-
-            switch (matchingParameter.Name)
+            // argument.Parameter will not be null here. The only case for it to be null is an __arglist argument, for which
```

Comment:
I think removing the check is fine. That said, I'm also not sure there's any harm leaving this simple check in as is :)

Reply:
Yeah there is no harm in keeping it. But the linked issue stated that it should either be deleted or a test coverage is added.